core/src/main/java/org/apache/brooklyn/core/mgmt/internal/LocationManagerInternal.java

```diff
@@ -19,10 +19,18 @@
 package org.apache.brooklyn.core.mgmt.internal;
 
 import org.apache.brooklyn.api.location.Location;
+import org.apache.brooklyn.api.location.LocationSpec;
 import org.apache.brooklyn.api.mgmt.LocationManager;
+import org.apache.brooklyn.config.ConfigKey;
+import org.apache.brooklyn.core.config.ConfigKeys;
 
 public interface LocationManagerInternal extends LocationManager, BrooklynObjectManagerInternal<Location> {
 
+    /** Indicates that a call to {@link #createLocation(LocationSpec)} should create a location instance not
+     * linked to the management context. This allows clients to peek at the result but not to use it. */
+    public static final ConfigKey<Boolean> CREATE_UNMANAGED = ConfigKeys.newBooleanConfigKey("brooklyn.internal.location.createUnmanaged",
```

Comment:
Yes we do in a few places, and probably will always, because some info is only available on the Location, not on the LocationSpec, and callers may want to know that.  (I suppose those places could unmanage it explicitly but why the overhead.)